src/oom/oomd-manager.c

```diff
@@ -378,17 +378,29 @@ static int monitor_cgroup_contexts_handler(sd_event_source *s, uint64_t usec, vo
                         OomdCGroupContext *t;
 
                         SET_FOREACH(t, targets) {
-                                log_notice("Memory pressure for %s is greater than %lu for more than %"PRIu64" seconds and there was reclaim activity",
-                                        t->path, LOAD_INT(t->mem_pressure_limit), m->default_mem_pressure_duration_usec / USEC_PER_SEC);
+                                _cleanup_free_ char *selected = NULL;
 
-                                r = oomd_kill_by_pgscan_rate(m->monitored_mem_pressure_cgroup_contexts_candidates, t->path, m->dry_run);
+                                log_debug("Memory pressure for %s is %lu.%02lu%% > %lu.%02lu%% for > %"PRIu64" seconds "
+                                          "with reclaim activity",
+                                          t->path, LOAD_INT(t->memory_pressure.avg10), LOAD_FRAC(t->memory_pressure.avg10),
+                                          LOAD_INT(t->mem_pressure_limit), LOAD_FRAC(t->mem_pressure_limit),
+                                          m->default_mem_pressure_duration_usec / USEC_PER_SEC);
+
+                                r = oomd_kill_by_pgscan_rate(m->monitored_mem_pressure_cgroup_contexts_candidates, t->path, m->dry_run, &selected);
                                 if (r == -ENOMEM)
                                         return log_oom();
                                 if (r < 0)
                                         log_notice_errno(r, "Failed to kill any cgroup(s) under %s based on pressure: %m", t->path);
                                 else {
                                         /* Don't act on all the high pressure cgroups at once; return as soon as we kill one */
                                         m->post_action_delay_start = usec_now;
+                                        if (selected != NULL)
+                                                log_notice("Killed %s due to memory pressure for %s being %lu.%02lu%% > "
+                                                           "%lu.%02lu%% for > %"PRIu64" seconds with reclaim activity",
+                                                           selected, t->path,
+                                                           LOAD_INT(t->memory_pressure.avg10), LOAD_FRAC(t->memory_pressure.avg10),
+                                                           LOAD_INT(t->mem_pressure_limit), LOAD_FRAC(t->mem_pressure_limit),
+                                                           m->default_mem_pressure_duration_usec / USEC_PER_SEC);
```

Comment:
one of these days this should become a structured log message really, i.e. log_struct() and get a MESSAGE_ID and a log catalog entry I am sure